### PR TITLE
Issue/293/cosmology object

### PR DESCRIPTION
**Pull request by pennalima**
CLMM cosmology object created, proper implementations of the backends, modified unit convention (M_sun, Mpc - no 'h'), tests of modeling and cosmology objects (work in progress #175). See issues #293 and #292. This branch also fixes issue #310.

### CODE REVIEW

**Comment by combet** [comment]
I went through all the notebooks to update the cosmology objects and modify units whenever necessary. I hope I didn't miss any, but this need to be carefully checked. As far as I can tell, all a running and giving expected results with all the backends.

Also, I came across an ValueError from scipy's `binned_statistic` in one of the notebook. This  was cause by an `inf` value in the values to binned. Never came across it before but it looks like this behaviour is a ['feature'](https://github.com/scipy/scipy/issues/11365) of `binned_statistic`, introduced in scipy v1.4.0 and I am currently running v1.4.1. So I went ahead and slighlty modified the `compute_radial_averages` in `utils.py` to filter out non-finite value before performing the binning. Would be good to have another pair of eyes checking this out as well.

File: examples/modeling_cmp_numcosmo.ipynb
Comment (aimalz) [inline examples/modeling_cmp_numcosmo.ipynb:236]: Should we clean out the output here, too?

**Review by aimalz — APPROVED** [review]
This is super!  Thanks especially for propagating the changes through the demos and tests, and writing the new tests.  I know there might be some more additions before folks are ready for a merge, but I'm going to approve it now on the basis that the changes proposed in #293 are complete and everything is organized consistently with the modeling/OO changes from #282/#299.

**Comment by combet** [comment]
A few small things I noticed looking at the API documentation:
- The API docs have a `clmm.modeling` module and a` clmm.clmm_modeling` module - first one corresponds to the way we call things in the functional interface, the latter to the OO interface. This may be confusing to the user?

Doctstrings to fix (units essentially), given what I've seen in the API documentation
- In the `clmm_cosmo` module:
    - [ ] Missing right-hand term for the `da` equation. Do we actually need an equation there?
    - [ ] `eval_da_a1a2` should return Mpc, not Mpc/h
- `clmm.clmm_modeling` module:
    - [ ] `eval_density` - correct units to Msun and Mpc only
    - [ ] `eval_shear` - Mpc only
    - [ ] `eval_sigma_crit` - Msun / Mpc2
    - [ ] there are not docstrings for the `eval_sigma_mean`, `set_concentration`, `set_cosmo` or `set_halo_density_profile` functions
    - [ ] `set_mass` - Msun, not Msun/h
- in the `clmm.modeling` module
    - [ ] `get_critical_surface_density` - return Msun/Mpc2

**Comment by combet** [comment]
Thanks for the docstring update @m-aguena. I've just fixed the remaining 'Unexpected indentation' error that was preventing the docs to build and causing Travis to fail. Looks like the equation need a line jump before and after them...

**Comment by m-aguena** [comment]
thanks @combet . I'd say this PR is ready for review then!

**Comment by combet** [comment]
@rherbonnet @marina-ricci, this PR is now ready for review. It would be great if we could get your feedback before tag up on Friday 23 Nov!

**Pull request merged**
